File: Sorting-1/insertionSort/InsertionSort.java
import  java.io.*;
import  java.util.*;
class InsertionSort{
    public static void insertionSort(int[] arr){
        for(int i=1;i<arr.length;i++){
            int j =i;
            while(j>0&&arr[j]<arr[j-1]){
                int temp = arr[j];
                arr[j]=arr[j-1];
                arr[j-1]=temp;
                j--;
            }
        }
    }

    public static void main(String[] args ){
        int arr[] = {20,40,11,19,27,45};
        insertionSort(arr);
        System.out.println(Arrays.toString(arr));

    }
}

File: Sorting-1/insertionSort/Js/insertionSort.js
function insertioSort(arr){
    for(let i = 1;i<arr.length;i++){
        let element = arr[i];
        let j;
        for( j = i-1;j>=0;j--){
            if(arr[j]>element){
                arr[j+1]=arr[j];
            }else{
                break;
            }
        }
        arr[j+1]=element;
    }
}

let arr = [5,4,3,1]
insertioSort(arr);
console.log(arr);
